zebrad/tests/acceptance.rs

```diff
@@ -1052,3 +1056,161 @@ async fn tracing_endpoint() -> Result<()> {
+    config.tracing.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened tracing endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `tracing.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, "(already in use)|(one usage)")?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same state directory, but different Zcash
+/// listener ports. The first node should get exclusive access to the database.
+/// The second node will panic with the Zcash state conflict hint added in #1535.
+#[test]
+fn zcash_state_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // A persistent config has a fixed temp state directory, but asks the OS to
+    // automatically choose an unused port
+    let mut config = persistent_test_config()?;
+    let dir_conflict = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(
+        dir_conflict.path(),
+        "Opened Zebra state cache",
+        dir_conflict.path(),
+        "lock file",
+    )?;
+
+    Ok(())
+}
+
+/// Launch a node in `first_dir`, wait a few seconds, then launch a node in
+/// `second_dir`. Check that the first node's stdout contains
+/// `first_stdout_regex`, and the second node's stderr contains
+/// `second_stderr_regex`.
+fn check_config_conflict<T, U>(
+    first_dir: T,
+    first_stdout_regex: &str,
+    second_dir: U,
+    second_stderr_regex: &str,
+) -> Result<()>
+where
+    T: ZebradTestDirExt,
+    U: ZebradTestDirExt,
+{
+    // Only execute this test in windows or linux families
+    if !cfg!(target_os = "macos") {
+        // Start the first node
+        let mut node1 = first_dir.spawn_child(&["start"])?;
+
+        // Wait a bit to spawn the second node, we want the first fully started.
+        std::thread::sleep(LAUNCH_DELAY_BIG);
+
+        // Spawn the second node
+        let mut node2 = second_dir.spawn_child(&["start"])?;
+
+        // Wait a few seconds and kill both nodes
+        std::thread::sleep(LAUNCH_DELAY_BIG);
+
+        node1.kill()?;
+        node2.kill()?;
+
+        // In node1 we want to check for the success regex
+        let output1 = node1.wait_with_output()?;
+        output1.stdout_contains(first_stdout_regex)?;
+        output1
+            .assert_was_killed()
+            .wrap_err("Possible port conflict. Are there other acceptance tests running?")?;
+
+        // In the second node we look for the conflict regex
+        let output2 = node2.wait_with_output()?;
+        output2.stderr_contains(second_stderr_regex)?;
+
+        // Windows is marking the exit as killed
+        if !cfg!(windows) {
```

Comment:
Why does Windows mark the exit as killed?
That doesn't happen in the rest of our tests.

is Windows hanging here?